source/contributing/implementation/datamanipulator.rst

```diff
@@ -0,0 +1,520 @@
+=============================
+Implementing DataManipulators
+=============================
+
+This is a guide for contributors who want to help with Data API implementation by creating DataManipulators.
+An updated list of DataManipulators to be implemented can be found at
+`SpongeCommon Issue #8 <https://github.com/SpongePowered/SpongeCommon/issues/8>`_.
+
+To fully implement a ``DataManipulator`` these steps must be followed:
+
+1. Implement the ``DataManipulator`` itself
+#. Implement the ``ImmutableDataManipulator``
+
+When these steps are complete, the following must also be done:
+
+3. Register the ``Key`` in the ``KeyRegistry``
+#. Implement the ``DataProcessor``
+#. Implement the ``ValueProcessor`` for each value being represented by the ``DataManipulator``
+#. Register everything in the ``SpongeSerializationRegistry``
+
+.. note::
+    Make sure you follow our :doc:`../guidelines`.
+
+1. Implement the DataManipulator
+================================
+
+The naming convention for ``DataManipulator`` implementations is the name of the interface prefixed with "Sponge".
+So to implement the ``HealthData`` interface, we create a class named ``SpongeHealthData`` in the appropriate package.
+For implementing the ``DataManipulator`` first have it extend an appropriate abstract class from the
+``org.spongepowered.common.data.manipulator.mutable.common`` package. The most generic there is ``AbstractData``
+but there are also abstractions that reduce boilerplate code even more for some special cases like
+``DataManipulator``\ s only containing a single value.
+
+.. code-block:: java
+
+    public class SpongeHealthData extends AbstractData<HealthData, ImmutableHealthData> implements HealthData {
+        [...]
+    }
+
+There are two type arguments to the AbstractData class. The first is the interface implemented by this class, the
+second is the interface implemented by the corresponding ``ImmutableDataManipulator``.
+
+The Constructor
+~~~~~~~~~~~~~~~
+
+In most cases while implementing an abstract Manipulator you want to have two constructors:
+
+* One without arguments (no-args) which calls the second constructor with "default" values
+* The second constructor that takes all the values it supports.
+
+The second constructor must
+
+* make a call to the ``AbstractData`` constructor, passing the class reference for the implemented interface.
+* make sure the values passed are valid
+* call the ``registerGettersAndSetters()`` method
+
+.. code-block:: java
+
+    public SpongeHealthData() {
+        this(20D, 20D);
+    }
+
+    public SpongeHealthData(double currentHealth, double maxHealth) {
+        super(HealthData.class);
+        Preconditions.checkArgument(currentHealth >= 0.0 && currentHealth <= (double) Float.MAX_VALUE);
+        Preconditions.checkArgument(maxHealth >= 0.0 && maxHealth <= (double) Float.MAX_VALUE);
```

Comment:
`Preconditions` is to be imported statically, but I'm not sure if/how you would want to represent this in a code snippet /shrug